### DISCUSSION TITLE
SIMPER analysis

### DISCUSSION THREAD

**Discussion by zhallber** [scikit-bio/scikit-bio#2106] [General]
Hi all -- I've been doing a lot of ordinations lately through scikit-bio, and many people have suggested that I try doing SIMPER analysis to find which taxa are driving variation along specific axes, but the only implementations I've been able to find are in R. Is there something similar that scikit-bio can do along these lines? Obviously don't want anyone to have to code anything new (although I won't say no), but I feel like I'm missing something in the code documentation that might work similarly.

Thanks!

**Comment by qiyunzhu** [comment]
Hello @zhallber Thank you for the suggestion. At this moment scikit-bio cannot do what `vegan::simper` does out of the box, although one may implement a homebrew solution by using the `samples` attribute of an [`OrdinationResults`](https://scikit.bio/docs/latest/generated/skbio.stats.ordination.html) instance, and utilizing the [`DistanceMatrix`](https://scikit.bio/docs/latest/generated/skbio.stats.distance.DistanceMatrix.html) class. We have a plan to significantly upgrade the ordination functionality in scikit-bio within the next several months. Your suggestion is valuable and I have forwarded it to our team.

**Reply by zhallber** [reply]
Hi @qiyunzhu thank you so much for the response and the idea on how to implement a homebrew solution! Just making sure that I wasn't missing anything that was named differently but still did something similar -- I really appreciate how well scikit-bio has worked for me, and major gratitude for how much energy and time coding all of these things take!!!
